contracts/swap/factory.go

```diff
@@ -0,0 +1,123 @@
+package swap
```

Comment:
Proposing to have the factory is a separate directory, as now the functions of `factory.go` are in the same namespace as `swap.go`, which is problematic because there are functions which overlap, such as `instanceAt` and `factoryAt` (which do essentially the same, so it would be better to name them the same as well).

Reply:
Not sure I agree, not a fan of one module per package....

Reply:
I would also keep them in the same package. They belong together anyway. Perhaps we should rename `instanceAt` to `chequebookAt`. Let's decide this on the rename PR.